tests/wrapper/test_bb_base.py

```diff
@@ -104,3 +105,56 @@ def test_multiple_wrappers(self):
         self.assertEqual(wrapper2._access_token, None)
 
         self.mocked_auth_requests.post.assert_called_once()
+
+    def test_authentication_fail_and_reauthentication(self):
+        """
+        Teste para verificar uma nova tentativa de autenticação
+        após uma falha na autenticação.
+
+        Dado:
+            -
+        Quando:
+            - for chamado BaseBBWrapper()._BaseBBWrapper__authenticate()
+        Então:
+            - 2 tentativas de autenticação devem ser realizadas,
+              uma falha e uma bem sucedida
+            - o resultado da autenticação deve ser True
+        """
+        bb_wrapper = BaseBBWrapper()
+
+        fail_attempts = 1
+        total_attempts = fail_attempts + 1
+
+        self.set_fail_auth(fail_attempts)
+
+        result = bb_wrapper._BaseBBWrapper__authenticate()
+
+        self.assertTrue(result)
+        self.assertEqual(total_attempts, self.mocked_auth_requests.post.call_count)
```

Comment:
Pode dar um assert no nº do token tbm, por causa da maneira que vc utilizou o call_count :clap: 

```suggestion
        self.assertEqual(total_attempts, self.mocked_auth_requests.post.call_count)
        self.assertEqual(bb_wrapper._token, 'token_2')
```